components/http_foundation.rst

```diff
@@ -585,6 +585,45 @@ represented by a PHP callable instead of a string::
 
 .. _component-http-foundation-serving-files:
 
+Streaming a JSON Response
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 6.2
+
+The :class:`Symfony\\Component\\HttpFoundation\\StreamedJsonResponse` class allows
+an API to return a lot of data as JSON and keep the used resources low by make usage
+of Generators.
+
+It expects a JSON structure with one or multiple replacers identifiers, as example
+the `'__articles__'`. As a second argument it requires one or multiple Generators
+which items can be converted to a JSON via ``json_encode`` method. The key of the
+Generators requires to be the used replacer identifiers::
+
+    use Symfony\Component\HttpFoundation\StreamedJsonResponse;
+
+    $response = new StreamedJsonResponse(
+        // json structure with replace identifiers
+        [
+            '_embedded' => [
+                'articles' => '__articles__',
+            ],
+        ],
+        // array of generators with replace identifier used as key
+        [
+            '__articles__' => (function (): \Generator { // any method or function returning a Generator
+                 yield ['title' => 'Article 1'];
+                 yield ['title' => 'Article 2'];
+                 yield ['title' => 'Article 3'];
+            })(),
+        ]
+    );
+
+.. tip::
+
+    If loading data via doctrine the ``toIterable`` method of ``Doctrine`` can be
+    used to keep also the resources low and fetch only one row one by one.
```

Comment:
```suggestion
    used to keep also the resources low and fetch only row by row.
```
?

Reply:
changed the text. I think it best to wait until #44709 discussion are finished to update the docs now